classes/controller.php

```diff
@@ -1266,62 +1266,56 @@ protected function taskDelmedia()
-                    ];
-                } else {
-                    $this->admin->json_response = [
-                        'status'  => 'error',
-                        'message' => $this->admin->translate('PLUGIN_ADMIN.FILE_COULD_NOT_BE_DELETED') . ': ' . $filename
-                    ];
-                }
-            } else {
-                //Try with responsive images @1x, @2x, @3x
-                $ext = pathinfo($targetPath, PATHINFO_EXTENSION);
-                $fullPathFilename = $page->path() . '/' . basename($targetPath, ".$ext");
-                $responsiveTargetPath = $fullPathFilename . '@1x.' . $ext;
-
-                $deletedResponsiveImage = false;
-                if (file_exists($responsiveTargetPath) && unlink($responsiveTargetPath)) {
-                    $deletedResponsiveImage = true;
-                }
+        if (!$filename) {
+            $this->admin->json_response = [
+                'status'  => 'error',
+                'message' => $this->admin->translate('PLUGIN_ADMIN.NO_FILE_FOUND')
+            ];
 
-                $responsiveTargetPath = $fullPathFilename . '@2x.' . $ext;
-                if (file_exists($responsiveTargetPath) && unlink($responsiveTargetPath)) {
-                    $deletedResponsiveImage = true;
-                }
+            return false;
+        }
 
-                $responsiveTargetPath = $fullPathFilename . '@3x.' . $ext;
-                if (file_exists($responsiveTargetPath) && unlink($responsiveTargetPath)) {
-                    $deletedResponsiveImage = true;
-                }
+        $targetPath = $page->path() . '/' . $filename;
 
-                if ($deletedResponsiveImage) {
-                    $this->admin->json_response = [
-                        'status'  => 'success',
-                        'message' => $this->admin->translate('PLUGIN_ADMIN.FILE_DELETED') . ': ' . $filename
-                    ];
-                } else {
-                    $this->admin->json_response = [
-                        'status'  => 'error',
-                        'message' => $this->admin->translate('PLUGIN_ADMIN.FILE_NOT_FOUND') . ': ' . $filename
-                    ];
-                }
+        if (!file_exists($targetPath)) {
+            $this->admin->json_response = [
+                'status'  => 'error',
+                'message' => $this->admin->translate('PLUGIN_ADMIN.FILE_NOT_FOUND') . ': ' . $filename
+            ];
 
-            }
+            return false;
+        }
+
+        $fileParts = pathinfo($filename);
+        $deleteCount = 0;
+
+        if (unlink($targetPath)) {
+            $deleteCount++;
         } else {
             $this->admin->json_response = [
                 'status'  => 'error',
-                'message' => $this->admin->translate('PLUGIN_ADMIN.NO_FILE_FOUND')
+                'message' => $this->admin->translate('PLUGIN_ADMIN.FILE_COULD_NOT_BE_DELETED') . ': ' . $filename
             ];
+
+            return false;
```

Comment:
Given the previous behavior, I'm not sure whether it's best to return here or to try and find and delete any image alternatives first. Some users may be stuck in a situation where they previously have tried to delete an image, and it wouldn't go away because it had an image alternative named `@4x` or higher. If we opt for this way of doing it, those files won't go away even if this patch is merged.

Probably an edge case, but maybe it wouldn't hurt to search for image alternatives in any case.